src/coreclr/src/jit/morph.cpp

```diff
@@ -13648,18 +13651,24 @@ GenTree* Compiler::fgMorphSmpOp(GenTree* tree, MorphAddrContext* mac)
                 break;
             }
 
-            bool foldAndReturnTemp;
+            bool      foldAndReturnTemp;
+            bool      isStore;
+            var_types addCastToType;
+
             foldAndReturnTemp = false;
-            temp              = nullptr;
-            ival1             = 0;
+            isStore           = (tree->gtFlags & GTF_DONT_CSE) == GTF_DONT_CSE;
```

Comment:
what would happen if we have `GTF_DONT_CSE` set but it is not LHS of a store? would it be a worse CQ or bad codegen?

Reply:
Currently there are a couple of places that are making this assumption.
We would have to change all of them to use GTF_IND_ASG_LHS and set that flag when morphing an assignment.